src/main/java/com/minecolonies/coremod/tileentities/TileEntityColonyBuilding.java

```diff
@@ -336,6 +341,12 @@ public void update()
                 colonyId = tempColony.getID();
             }
         }
+
+        /*
+         * We want a new inventory every tick.
+         * The accessed inventory in the same tick must be the same.
+         */
+        combinedInv = null;
```

Comment:
(This is also the answer to the question below):
In many queries we query in the same tick the capability of this container for all enum facing + null. Since it creates a new combined inventory each time, the set will add 7 inventories in it. Which costs performance and makes the results wrong.
I, therefore, added that it caches the combined inventory.
But, since the inventory of the building can change, when new containers are added or removed. I update this here regularly so its only the same during the same tick.